frontend/src/libs/components/dropdown/dropdown.tsx

```diff
@@ -1,96 +1,64 @@
 import {
   type Control,
+  type ControllerRenderProps,
   type FieldErrors,
   type FieldPath,
   type FieldValues,
 } from 'react-hook-form';
 import Select, {
+  type ClassNamesConfig,
   type GroupBase,
   type SingleValue,
-  type StylesConfig,
 } from 'react-select';
 
 import { getValidClassNames } from '~/libs/helpers/helpers.js';
-import {
-  useCallback,
-  useFormController,
-  useMemo,
-  useState,
-} from '~/libs/hooks/hooks.js';
+import { useCallback, useMemo, useState } from '~/libs/hooks/hooks.js';
 import { type SelectOption } from '~/libs/types/select-option.type.js';
 
 import styles from './styles.module.scss';
 
 type Properties<T extends FieldValues> = {
   options: SelectOption[];
   name?: FieldPath<T>;
-  control?: Control<T, null>;
+  control?: Control<T, null> | undefined;
   errors?: FieldErrors<T>;
   label?: string;
   defaultValue?: SelectOption;
-  onChange?: (value: string | undefined) => void;
+  onChange?: (option: SingleValue<SelectOption>) => void;
+  placeholder?: string;
+  field?: ControllerRenderProps<T, FieldPath<T>>;
+  className?: string;
 };
 
-const getStyles = (
+const getClassNames = (
   isMenuOpen: boolean,
-): StylesConfig<SelectOption, false, GroupBase<SelectOption>> => {
-  return {
-    control: (styles) => ({
-      ...styles,
-      height: '20px',
-      minHeight: '20px',
-      border: 'none',
-      boxShadow: 'none',
-      outline: 'none',
-    }),
-    option: (styles) => ({
-      ...styles,
-      padding: '10px',
-      margin: '-4px 0',
-    }),
-    valueContainer: (styles) => ({
-      ...styles,
-      padding: '0',
-      paddingLeft: '5px',
-    }),
-    dropdownIndicator: (styles) => ({
-      ...styles,
-      transition: 'all 0.2s ease',
-      margin: '0',
-      padding: '0',
-      transform: `scaleY(${isMenuOpen ? '-1' : '1'})`,
-    }),
-    indicatorSeparator: () => ({
-      display: 'none',
-    }),
-    menu: (styles) => ({
-      ...styles,
-      overflow: 'hidden',
-      padding: 0,
-    }),
-  };
-};
+): ClassNamesConfig<SelectOption, false, GroupBase<SelectOption>> => ({
+  container: () => styles.container,
+  control: () => styles.control,
+  option: () => styles.option,
+  menu: () => styles.singleValue,
+  placeholder: () => styles.placeholder,
+  singleValue: () => styles.singleValue,
+  valueContainer: () => styles.valueContainer,
+  dropdownIndicator: () =>
+    isMenuOpen
+      ? getValidClassNames(styles.dropdownIndicator, styles.upside)
+      : getValidClassNames(styles.dropdownIndicator),
```

Comment:
you don't need getValidClassNames with 1 class

Reply:
fixed